src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -136,17 +134,43 @@ mode with existing flows and hold flows of different table-mode types"() {
 
         then: "Flow remains valid and pingable, switch validation passes"
         northbound.validateFlow(flow.flowId).each { assert it.asExpected }
+        "validate switches"(involvedSwitches)
+        assert "is flow pingable?"(flow), FLOW_ISNT_PINGABLE_ASSERTION_MESSAGE
+
+        and: "Flow allows traffic"
+        assert "does flow pass traffic?"(flow), FLOW_DOESNT_PASS_TRAFFIC_ASSERTION_MESSAGE
+
+        when: "Delete flow"
+        def flowCookies = getFlowCookies(swPair, flow)
+        flowHelperV2.deleteFlow(flow.flowId)
+        def flowIsDeleted = true
+
+        then: "Flow rules are deleted from switches"
         involvedSwitches.each { sw ->
-            with(northbound.validateSwitch(sw.dpId)) { validation ->
-                validation.verifyRuleSectionsAreEmpty(["missing", "excess", "misconfigured"])
-                validation.verifyHexRuleSectionsAreEmpty(["missingHex", "excessHex", "misconfiguredHex"])
-                validation.verifyMeterSectionsAreEmpty(["missing", "excess", "misconfigured"])
+            with(northbound.getSwitchRules(sw.dpId).flowEntries) { rules ->
+                rules.findAll { it.cookie in flowCookies }.empty
             }
         }
-        verifyAll(northbound.pingFlow(flow.flowId, new PingInput())) {
-            it.forward.pingSuccess
-            it.reverse.pingSuccess
-        }
+
+        cleanup: "Revert system to original state"
+        !flowIsDeleted && flow && flowHelperV2.deleteFlow(flow.flowId)
+        initSwProps && revertSwitchesToInitState(involvedSwitches, initSwProps)
+        northbound.deleteLinkProps(northbound.getLinkProps(topology.isls))
+    }
+
+    @Tidy
+    @Tags([LOW_PRIORITY])
+    def "System can hold flows of different table-mode types"() {
+        given: "A potential flow on a path of 4 switches: multi -> single -> multi -> single"
+        def (involvedSwitches, swPair, desiredPath, initSwProps) =
+        "get a potential flow on a path of 4 switches: multi and single"()
+        when: "Create the prepared hybrid flow"
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+        assert PathHelper.convert(northbound.getFlowPath(flow.flowId)) == desiredPath
+
+        then: "Created flow is valid"
+        northbound.validateFlow(flow.flowId).each { assert it.asExpected }
```

Comment:
CRITICAL: Before creating second flow you must change mode of switches with

`when: "Update table mode for involved switches so that it becomes 'single -> multi -> single -> multi'"`

Because name of the test is `System can hold flows of different table-mode types` but you have 2 flows with same table mode types

Reply:
Well, test name is obviously ambiguous. First of all, if I get it right, only switches can be in multi/single table mode, not flows. And idea behind the test is to check if there can be flow which goes through both single and multi table mode.
Second, if the deal is only about checking if the first flow rules can be placed in several tables, and the rules of the second flow can be in table '0', why do we need four switches for verification?
Do you have a guess what this test was expected to verify?

Reply:
Flow can be it single or multi mode too. We nad to do it because of some restrictions. To change mode of flow you need to
1. updated switch mode
2. reroute flow.

The this test case must check that two flows in different modes can live on one switch. 

How it works:
You find switches with following modes
`multi -> single -> multi -> single`

You create a flow. it has following modes on switches
`multi -> single -> multi -> single`

You change switch modes to 
`single -> multi -> single -> multi`
but first flow still has previous mode because nobody rerouted/updated it

you create second flow and get two flows with following modes

first:
`multi -> single -> multi -> single`
second:
`single -> multi -> single -> multi`

Reply:
Ok, I've got the picture. I think we don't need 4 switches to verify this option then. It will be enough to have two:
1. single <-> single
2. add flow
3. multi-multi
4. add 2nd flow
Because we've already verified that flows work ok on path single -> multi -> single...

Reply:
No, we still need 3-4 switches. 
flow has ingress switch (src), transit switch (middle), egress switch (dst). 
Kilda install different rules for ingress/egress/transit switches so we have to check all cases.

Reply:
Updated this test for 3-switches setup + added test for one-switch flow rerouting.